sno/conflicts.py

```diff
@@ -162,211 +159,74 @@ def _ensure_conflict(self, conflict):
-        )
-        dataset = first_true(datasets3)
-        dataset_path = dataset.path
-        if None in datasets3:
-            for cwr, ds in zip(commit_with_refs3, datasets3):
-                presence = "present" if ds is not None else "absent"
-                click.echo(f"{cwr}: {dataset_path} is {presence}")
-            raise NotYetImplemented(
-                "Sorry, resolving conflicts where features are added or removed isn't supported yet"
-            )
+    for key, conflict3 in conflict_index.conflicts.items():
+        paths3 = aot(c.path if c else None for c in conflict3)
+        if not any(paths3):
+            # Shouldn't happen
+            raise RuntimeError("Conflict has no paths")
 
-        pks3 = aot(
-            ds.index_entry_to_pk(ie) for ds, ie in zip(datasets3, index_entries3)
-        )
-        if "META" in pks3:
-            click.echo(f"Merge conflict found in metadata for {dataset_path}")
-            raise NotYetImplemented(
-                "Sorry, resolving conflicts in metadata isn't supported yet"
-            )
-        pk = first_true(pks3)
-        if pks3.count(pk) != 3:
-            click.echo(
-                f"Merge conflict found where primary keys have changed in {dataset_path}"
-            )
-            for cwr, pk in zip(commit_with_refs3, pks3):
-                click.echo(f"{cwr}: {dataset_path}:{pk}")
-            raise NotYetImplemented(
-                "Sorry, resolving conflicts where primary keys have changed isn't supported yet"
-            )
-        conflict_pks.setdefault(dataset_path, [])
-        conflict_pks[dataset_path].append(pk)
-
-    num_conflicts = sum(len(pk_list) for pk_list in conflict_pks.values())
-    click.echo(f"\nFound {num_conflicts} conflicting features:")
-    for dataset_path, pks in conflict_pks.items():
-        click.echo(f"{len(pks)} in {dataset_path}")
-    click.echo()
-
-    # Check for dirty working copy before continuing - we don't want to fail after interactive part.
-    ours_rs = repo_structures3.ours
-    ours_rs.working_copy.reset(commits3.ours, ours_rs)
-
-    # At this point, the failure should be dealt with so we can start resolving conflicts interactively.
-    # We don't want to fail during conflict resolution, since then we would lose all the user's work.
-    # TODO: Support other way(s) of resolving conflicts.
-    input_mode = get_input_mode()
-    if dry_run:
-        click.echo("Printing conflicts but not resolving due to --dry-run")
-    elif input_mode == InputMode.INTERACTIVE:
-        interactive_pause(
-            "Press enter to begin resolving merge conflicts, or Ctrl+C to abort at any time..."
-        )
-    elif input_mode == InputMode.NO_INPUT:
-        click.echo(
-            "Printing conflicts but not resolving - run from an interactive terminal to resolve"
-        )
+        tables3 = aot(p.split("/.sno-table/", 2)[0] if p else None for p in paths3)
```

Comment:
could you comment what these paths look like?
should `p.split("/.sno-table/", 2)` be `p.split("/.sno-table/", 1)` ?

Reply:
Done